typing/includecore.ml

```diff
@@ -169,26 +169,20 @@ let value_descriptions ~loc env name
   | Val_prim p1 -> begin
      match vd2.val_kind with
      | Val_prim p2 -> begin
-         let restrict_locality = [
-           (fun m -> Mode.Locality.submode_exn m Mode.Locality.global);
-           (fun m -> Mode.Locality.submode_exn Mode.Locality.local m)
-         ] in
-         let restrict_yielding = [
-           (fun m -> Mode.Yielding.submode_exn m Mode.Yielding.unyielding);
-           (fun m -> Mode.Yielding.submode_exn Mode.Yielding.yielding m)
-         ] in
-         List.iter (fun restrict_loc ->
-           List.iter (fun restrict_yield ->
+         let locality = [ Mode.Locality.global; Mode.Locality.local ] in
+         let yielding = [ Mode.Yielding.unyielding; Mode.Yielding.yielding ] in
```

Comment:
Sorry to keep pushing this, then now I think about this, it might be even better to have e.g. `Mode.Locality.Const.all` which is a list of all the constants? I'm slightly worried that we might introduce more modes in the axis and we forgot to update the check here.